101-bundle-json.md

```diff
@@ -268,37 +268,47 @@ Fields:
   - `digest`: MUST contain a digest, in [OCI format](https://github.com/opencontainers/image-spec/blob/master/descriptor.md#digests), to be used to compute the integrity of the image. The calculation of how the image matches the digest is dependent upon image type. (OCI, for example, uses a Merkle tree while VM images are checksums.)
-  - `refs`: An array listing the locations which refer to this image, and whose values should be replaced by the value specified in URI. Each entry contains the following properties:
+  - `refs`: An array listing the locations which refer to this image, and whose values should be replaced by the value specified in URI. The behavior associated with this section is defined below. Each entry contains the following properties:
     - `path`: The path of the file where the value should be replaced
     - `field`: A selector specifying a location (or locations) within that file where the value should be replaced
     - `mediaType`: The media type of the file, which can be used to determine the file type. If unset, tooling may choose any strategy for detecting format
+    - `expressionType`: The expression language that should be used to evaluate `field`.
   - `size`: The image size in bytes
   - `platform`: The target platform, as an object with two fields:
     - `architecture`: The architecture of the image (`i386`, `amd64`, `arm32`...)
     - `os`: The operating system of the image
   - `mediaType`: The media type of the image
 
-Substitutions MUST be supported for the following formats:
+### Reference Substitution (`refs`)
 
-- JSON
-- YAML
-- XML
+The `refs` section provides tools for mapping fully qualified image names into paths present within the invocation image.
 
-In addition to these substitutions, the image map data is also made available to the invocation image at runtime. See [Image map](103-bundle-runtime.md#image-map) for more details.
+A CNAB Runtime MUST pass the images map into the invocation image, as described in [the Bundle Runtime definition](103-bundle-runtime.md#image-map). Invocation images MAY implement support for this feature.
 
-### Field Selectors
+For example, consider the following `refs` declaration:
 
-*TODO:* We have multiple competing standards in this space, and those that are popular for JSON are not the same as those popular for XML. This portion is thus not complete.
-
-For fields, the selectors are based on the _de facto_ format used in tools like `jq`, which is a subset of the [CSS selector](https://www.w3.org/TR/selectors-3/) path. Examples:
+```json
+"images": {
+    "my-microservice": {
+      "description": "my microservice",
+      "digest": "sha256:aaaaaaaaaaaa...",
+      "image": "technosophos/microservice:1.2.3",
+      "refs": [
+        {
+          "field": "service.image",
+          "path": "/cnab/charts/values.yaml",
+          "expressionType": `css-selector`
```

Comment:
I was slightly confused to see the expressionType of `css-selector` in this example for setting a chart value.  Was this intentional?  Or should it be `pcre`?